# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1532689367 14400
#      Fri Jul 27 07:02:47 2018 -0400
# Node ID a4b3b6591c239f699f84e905a96eb79a04c2e03c
# Parent  b2917b7858ba02c08c880f3dae679fc51d535632
# Parent  ff473a4187e5c622979afc93b81f5258ec41933d
maint: Merge stable to default.

diff --git a/liboctave/wrappers/fpucw-wrappers.c b/liboctave/wrappers/fpucw-wrappers.c
--- a/liboctave/wrappers/fpucw-wrappers.c
+++ b/liboctave/wrappers/fpucw-wrappers.c
@@ -50,22 +50,26 @@ octave_set_default_fpucw (void)
 {
   fpucw_t cw = GET_FPUCW ();
 
   if (cw != _FPU_DEFAULT)
     SET_FPUCW (_FPU_DEFAULT);
 }
 
 // OLDCW is the name used by the fpucw.h macros and gnulib doesn't give
-// us a choice.  We are also assuming that fpucw_t is unsigned int.
+// us a choice.  We are also assuming that fpucw_t is unsigned, and no
+// wider than an int.
 
 unsigned int
 octave_begin_long_double_rounding (void)
 {
-  DECL_LONG_DOUBLE_ROUNDING
+  // Don't use DECL_LONG_DOUBLE_ROUNDING here because on some systems,
+  // it is defind to be empty.
+
+  fpucw_t oldcw = 0;
 
   BEGIN_LONG_DOUBLE_ROUNDING ();
 
   return oldcw;
 }
 
 void
 octave_end_long_double_rounding (unsigned int oldcw)
